akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -36,20 +36,40 @@ trait RetrySupport {
    * }}}
    */
   def retry[T](attempt: () ⇒ Future[T], attempts: Int, delay: FiniteDuration)(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] = {
-    try {
-      if (attempts > 0) {
-        attempt() recoverWith {
-          case NonFatal(_) ⇒ after(delay, scheduler) {
-            retry(attempt, attempts - 1, delay)
+    retry(attempt, attempts, delay, (_, initialDelay, _) ⇒ initialDelay)
+  }
+
```

Comment:
add DoNotInherit to `trait RetrySupport`. At least that is what I think the intention is, otherwise we would not be able to add a new method like this.

```
 *
 * Not for user extension
 */
@DoNotInherit 
```

Reply:
and make it abstract class while you are at it, otherwise it will not be usable from Java when built with Scala 2.11

Reply:
and add Java API with CompletionStage and java.time.Duration if you like

Reply:
@patriknw It was mixed in to `package object pattern `  so can't be a class, otherwise it will be the only one Class that package object mixed in.